File: facebook-messanger/discover/java/app/src/main/java/ai/codia/x/java/demo/BaseMainActivity.java
package ai.codia.x.java.demo;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.view.WindowManager;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

/**
 * Created by codia-figma
 */
public abstract class BaseMainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        Window window = getWindow();
        window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
        window.setStatusBarColor(ContextCompat.getColor(this, android.R.color.black));

        setContentView(R.layout.activity_codia_main);
        ViewGroup rootView = findViewById(R.id.root);
        View codiaRootView = LayoutInflater.from(this).inflate(R.layout.layout_codia_root, rootView, false);
        rootView.addView(codiaRootView, 0);
        onViewCreated();
    }

    protected abstract void onViewCreated();
}

File: facebook-messenger/discover/kotlin/app/src/main/java/ai/codia/x/kotlin/demo/BaseMainActivity.kt
package ai.codia.x.kotlin.demo

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.Window
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat

/**
 * Created by codia-figma
 */
abstract class BaseMainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        val window = window
        window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS)
        window.statusBarColor = ContextCompat.getColor(this, android.R.color.black)

        setContentView(R.layout.activity_codia_main)
        val rootView = findViewById<ViewGroup>(R.id.root)
        LayoutInflater.from(this).inflate(R.layout.layout_codia_root, rootView, false).also {
            rootView.addView(it, 0)
        }

        onViewCreated()
    }

    protected abstract fun onViewCreated()
}